README.rst

```diff
@@ -387,6 +387,55 @@ include any additional headers.
         resp = session.send(prepped)
         assert resp.text == "hello world"
 
+Response Registry
+---------------------------
+
+By default, ``responses`` module will search all registered``Response`` objects and
+return a match. If only one ``Response`` is registered, the registry is kept unchanged.
+However, if multiple matches are found for the same request, then first match is returned and
+removed from registry.
+
+Such behavior is suitable for most of use cases, but to handle special conditions, you can
+implement custom registry which must follow interface of ``registries.FirstMatchRegistry``.
+Redefining the ``find`` method will allow you to create custom search logic and return
+appropriate ``Response``
+
+Example that shows how to set custom registry
+
+.. code-block:: python
+
+    import responses
+    from responses import registries
+
+
+    class CustomRegistry(registries.FirstMatchRegistry):
+        pass
+
+
+    print("Before tests:", responses.mock._get_registry())
```

Comment:
I don't think we should be giving examples of using non-public examples. The direction taken in #440 where there is a public `.registry` would let users who need to interact with the registry use public APIs.

Reply:
updated